packages/models-library/src/models_library/api_schemas_webserver/socketio.py

```diff
@@ -1,17 +1,16 @@
+from ..basic_types import IDStr
 from ..users import GroupID, UserID
 
 
-class SocketIORoom(str):
-    __slots__ = ()
-
+class SocketIORoomStr(IDStr):
     @classmethod
-    def from_socket_id(cls, socket_id: str) -> "SocketIORoom":
+    def from_socket_id(cls, socket_id: str) -> "SocketIORoomStr":
         return cls(socket_id)
```

Comment:
Note that `IDStr` is **not** a `BaseModel` but a constrainted type therefore the validators will not run using the constructor i.e. `cls(socket_id)` will not validate. You need either to explicitly call them here or use something like `parse_obj_as(cls, socket_id)` which will create a root `BaseModel` with `IDStr`